net.c

```diff
@@ -32,118 +210,28 @@ make_server_socket(char *host, char *port)
         fd = SD_LISTEN_FDS_START;
         r = sd_is_socket_inet(fd, 0, SOCK_STREAM, 1, 0);
         if (r < 0) {
-            errno = -r;
             twarn("sd_is_socket_inet");
+            errno = -r;
             return -1;
         }
         if (!r) {
```

Comment:
Are sure that r==0 here in the case when systemd provided listen fd for the unix socket?

Reply:
[From `libsystemd`'s man pages][1]:

> ### Return value
>
> On failure, these calls return a negative errno-style error code. If the file descriptor is of the specified type and bound to the specified address, a positive return value is returned, otherwise zero.

[1]: https://www.freedesktop.org/software/systemd/man/sd_is_socket_inet.html#

Reply:
Ah, so it is the same as with case for inet socket above. Sounds right then.